### DISCUSSION TITLE
Trying to run Dropbox

### DISCUSSION THREAD

**Discussion by michaeljh9** [ptitSeb/box86#511] [General]
Greetings. I'm running a Raspberry Pi 4 8GB RAM running 64-bit Ubuntu. My goal is to run Dropbox. I have successfully installed box86 using this guide:

https://pimylifeup.com/raspberry-pi-x86/

https://www.dropbox.com/install-linux is where I downloaded the 32-bit *.deb file. 

I ran 'box86 dropbox_2020.03.04_i386.deb' and got the error message
'dropbox_2020.03.04_i386.deb is not an executable file'

For my second attempt, I downloaded Dropbox from the 'compile from source' directory and downloaded nautilus-dropbox-2020.03.04.tar.bz2. I extracted it and ran

'box86 ./install-sh'

and got the error message 

Error: reading elf header of nautilus-dropbox-2020.03.04/install-sh, try to launch natively instead
Segmentation fault (core dumped)

What am I doing wrong? Do you have any suggestions?

**Comment by ptitSeb** [comment]
box86 allows you to run x86 binaries as if they where arm one. A ".deb" is a package used by debian and ubuntu to install software, it's not a binary. I don't advise using that directly (you would need to add i386 architecture to).

The `install-sh` is a bash shell, that contains script and archive with the binary. This kind of script has to be launched directly, or using `bash`.
In fact, if back86 is installed and configured in the system (using the `sudo make install`), you don't have to use box86, it will be used automaticaly and transparently (you'll just see a the "Box86 ..." banner when it's used).
Also, I suggest you install box64 on your system, in case dropbox use some 64bits binary?

I haven't tried Dropbox, so not sure it works. But if I remember correctly, you need gtk, so  you need `sudo apt install libgtk2.0-0:armhf` but also maybe some other libs. Use `BOX86_LOG=1 ./install-sh` to have more logs of what is happening.

**Comment by michaeljh9** [comment]
Hi ptitSeb:
I had no idea Box64 was available and wasn't aware of its availability. I want to give my Pi the best chance of success so I re-imaged Ubuntu and installed Box64.

Based on your advice, I installed libgtk2.0-0:armhf.
I ran `BOX86_LOG=1 ./install-sh` and got the error
`./install-sh: no input file specified`

For my second attempt, I followed the guide here:
https://help.dropbox.com/installs-integrations/desktop/linux-commands#build

and when their instructions say run `./configure`, what I actually ran was
`BOX64_LOG=1 ./configure`

The error generated was:

```
configure: error: Package requirements (libnautilus-extension >= 2.16.0) were not met:
No package 'libnautilus-extension' found
Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.
```

Am I getting further along? When you said Box runs x86 binaries, you make it sound like the Windows installers is what can be run. What am I missing?

**Comment by ptitSeb** [comment]
What I'm saying is that box86 is meant to runs programs. You can try to install the windows client, but you'll need (x86) Wine installed on your system for that.
"./configure" is meant for building. TBH, I haven't looked at what you grabbed. Looks like some sources, to be built directly (not using box86 here). But, if I remember well, Dropbox use some pre-built static library, so cannot be built on ARM or ARM64 machine. You nned to grab a pre-built binary (in the form of a tar.gz for example), but there is none.

I'll try later to give some direction on how to install it. But I need to test and all, will do that later...

**Comment by michaeljh9** [comment]
Let me know if I can provide any more useful info.
Depending on what box + wine guide I'm following, the source for installing wine differs from either getting it from Linux's built-in repositories or letting pi-apps take care of it. From the backend, is there a significant difference? Do you recommend one over the other?
